packages/astro/src/astro-components/unstyled/SignOutButton.astro

```diff
@@ -0,0 +1,27 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+type Props<Tag extends HTMLTag> = Polymorphic<{ as: Tag; id?: string; redirectUrl?: string }>
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
```

Comment:
Starting to notice we're repeating this piece of logic in all our Astro components. I'll make a ticket that will added this